src/app/groups/containers/user-group-invitations/user-group-invitations.component.ts

```diff
@@ -1,97 +1,122 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
-import { merge, of, Subject } from 'rxjs';
-import { switchMap, map } from 'rxjs/operators';
-import { GridColumn } from 'src/app/ui-components/grid/grid.component';
-import { displayResponseToast } from 'src/app/groups/containers/pending-request/pending-request-response-handling';
-import { fetchingState, readyState } from 'src/app/utils/state';
+import { ReplaySubject } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
 import { GetRequestsService, PendingRequest } from '../../data-access/get-requests.service';
-import { Action, parseGroupInvitationResults, RequestActionsService } from '../../data-access/request-actions.service';
 import { ActionFeedbackService } from 'src/app/services/action-feedback.service';
 import { HttpErrorResponse } from '@angular/common/http';
 import { CurrentContentService } from 'src/app/services/current-content.service';
 import { PendingRequestComponent } from '../pending-request/pending-request.component';
 import { LoadingComponent } from '../../../ui-components/loading/loading.component';
-import { NgIf } from '@angular/common';
+import { AsyncPipe, DatePipe, NgClass, NgForOf, NgIf, NgSwitch, NgSwitchCase, NgSwitchDefault } from '@angular/common';
 import { ErrorComponent } from '../../../ui-components/error/error.component';
+import { ProcessGroupInvitationService } from '../../data-access/process-group-invitation.service';
+import { ButtonModule } from 'primeng/button';
+import { TableModule } from 'primeng/table';
+import { UserCaptionPipe } from '../../../pipes/userCaption';
+import { SortEvent } from 'primeng/api/sortevent';
+import { mapToFetchState } from '../../../utils/operators/state';
 
 @Component({
   selector: 'alg-user-group-invitations',
   templateUrl: './user-group-invitations.component.html',
   styleUrls: [ './user-group-invitations.component.scss' ],
   standalone: true,
-  imports: [ PendingRequestComponent, LoadingComponent, NgIf, ErrorComponent ],
+  imports: [
+    PendingRequestComponent,
+    LoadingComponent,
+    NgIf,
+    ErrorComponent,
+    ButtonModule,
+    DatePipe,
+    NgForOf,
+    NgSwitchCase,
+    TableModule,
+    UserCaptionPipe,
+    NgSwitch,
+    NgSwitchDefault,
+    AsyncPipe,
+    NgClass,
+  ],
 })
-export class UserGroupInvitationsComponent implements OnDestroy, OnInit {
-  requests: PendingRequest[] = [];
-
-  readonly columns: GridColumn[] = [
-    { field: 'group.name', header: $localize`TITLE` },
-    { field: 'group.type', header: $localize`TYPE` },
-    { field: 'at', header: $localize`REQUESTED ON`, sortKey: 'at' },
-  ];
-
-  state: 'fetching' | 'processing' | 'ready' | 'fetchingError' = 'fetching';
+export class UserGroupInvitationsComponent implements OnInit, OnDestroy {
   currentSort: string[] = [];
+  processing = false;
 
-  private dataFetching = new Subject<{ sort: string[] }>();
+  private groupInvitationsSubject = new ReplaySubject<{ sort: string[] }>();
```

Comment:
What is this subject for ? Why isn't `currentSort` enough?

Reply:
Fixed